packages/components/src/DataList/DataList.utils.tsx

```diff
@@ -144,3 +164,88 @@ export function generateDataListEmptyState({
 
   return EmptyStateComponent;
 }
+
+export function renderDataListLayout<T extends DataListObject>(
+  layouts: React.ReactElement<DataListLayoutProps<T>>[] | undefined,
+  elementData: DataListItemType<T[]>[],
+) {
+  const sizePropsOfLayouts = layouts?.map(layout => layout.props.size || "xs");
+  return layouts?.map(layout => {
+    const layoutChildren = layout.props.children;
+    const layoutSize = layout.props.size || "xs";
+    const largerBreakpointsToHide = sizePropsOfLayouts?.filter(
+      size => BREAKPOINTS.indexOf(size) > BREAKPOINTS.indexOf(layoutSize),
+    );
+    const cssVars = getCSSVariablesForBreakpoints(
+      layoutSize,
+      largerBreakpointsToHide,
+    );
+    return elementData.map((child, i) => {
+      // TODO: Don't use index as key. Might have to force an ID on the data JOB-76773
+      return (
+        <div
+          className={styles.listItem}
+          key={`${i}-${layoutSize}`}
+          style={cssVars}
+        >
+          {layoutChildren(child)}
+        </div>
+      );
+    });
+  });
+}
+
+export function renderDataListHeader<T extends DataListObject>(
```

Comment:
Not a big fan of having so much duplicated code. I think it may be worth looking at once we add sorting support to see what can be done.

Reply:
Cleaned it up